### PR TITLE
Postings highlighter deprecation

### PR DESCRIPTION
**Pull request by jimczi**
The `postings` highlighter is deprecated in Lucene and will be replaced by the `unified`.
This change adds a deprecation warning for removal in 6.0.

Relates https://github.com/elastic/elasticsearch/pull/25028

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/PostingsHighlighter.java [hunk 1]
```diff
@@ -44,13 +46,15 @@
...
+@Deprecated
 public class PostingsHighlighter implements Highlighter {
+    private static DeprecationLogger deprecationLogger = new DeprecationLogger(Loggers.getLogger(PostingsHighlighter.class));
 
     private static final String CACHE_KEY = "highlight-postings";
 
     @Override
     public HighlightField highlight(HighlighterContext highlighterContext) {
-
+        deprecationLogger.deprecated("[postings] highlighter is deprecated, please use [unified] instead");
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/PostingsHighlighter.java]: this way we will issue a deprecation warning for each hit, each field. Should we try to do it for each document at least? or maybe we can do it for each request, that would be even better.

Comment (jimczi) [inline core/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/PostingsHighlighter.java]: Thanks, I tried to limit to one per request in this change:
https://github.com/elastic/elasticsearch/pull/25073/commits/31ee14516ae2fa1baa14089b62ed8018120e58e5
Can you take another look ?

File: docs/reference/search/request/highlighting.asciidoc [hunk 2]
```diff
@@ -60,8 +60,13 @@ This is repeated for every field and every document that needs highlighting. If
 [[postings-highlighter]]
 ==== Postings highlighter
 
+deprecated[5.5.0,This highlighter type is deprecated and will be removed. Please use the `unified` highlighter instead. ]
```
Comment (nik9000) [inline docs/reference/search/request/highlighting.asciidoc]: removed in 6.0.

**Review by nik9000 — APPROVED** [review]

**Comment by jimczi** [comment]
Thanks @nik9000 and @javanna

**Pull request merged**